content/src/content/jcr_root/apps/core/wcm/cloudservices/documentcloud/clientlib/js/documentcloud.js

```diff
@@ -0,0 +1,191 @@
+
+        if ($propertiesForm.length !== 0) {
+            $("#documentcloud-cloudconfiguration-name").prop("disabled", true);
+        }
+
+        var configAlreadyExists = function(configName, configContainer) {
+            var result = $.ajax({
+                type: "GET",
+                async: false,
+                url: Granite.HTTP.externalize(encodeURI(configContainer) + ".1.json"),
+                cache: false
+            });
+
+            if (result.status !== 200) {
+                return false;
+            }
+
+            if (result.responseText !== null && result.responseText !== "") {
+                var nodeList = JSON.parse(result.responseText);
+                for (var i in nodeList) {
+                    if (configName === i) {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        };
+
+        var displayError = function(errorMessage) {
+            var $errorElement = $("#documentcloud-cloudconfiguration-error-message");
+            $errorElement.text(errorMessage);
+
+            $("#documentcloud-common-dialog").adaptTo("foundation-toggleable").show();
+            $errorElement.removeAttr("hidden");
+            $("#documentcloud-cloudconfiguration-already-exist-message").attr("hidden");
+        };
+
+        var submitForm = function($form, url) {
+            $.ajax({
+                type: $form.prop("method"),
+                url: url,
+                data: $form.serialize(),
+                cache: false,
+                contentType: $form.prop("enctype")
+            }).done(function(data, textStatus, jqXHR) {
+                var configurationContainer = url.substring(url.indexOf("/conf/"), url.indexOf("/settings/"));
+                window.location.href =
+                    Granite.HTTP.externalize("/mnt/overlay/core/wcm/cloudservices/documentcloud.html" + configurationContainer);
+            }).fail(function(jqXHR, textStatus, errorThrown) {
+                displayError(errorThrown);
+            });
+        };
+
+        var isRestricted = function(charCode, keyCode) {
+            if (charCode === 0 && (keyCode === 8 || keyCode === 9 || keyCode > 36 && keyCode < 47)) { //37-46 insert/delete/arrow keys
+                return false;
+            }
+            if ((charCode > 47 && charCode < 58) ||  // 0-9 digits
+                (charCode > 64 && charCode < 91) ||  // A-Z
+                (charCode > 96 && charCode < 123) ||  // a-z
+                (charCode === 45) || (charCode === 95)) { // "-" and "_"
+                return false;
+            } else {
+                return true;
+            }
+        };
+
+        var replaceRestrictedCodes = function(value) {
+            if (value) {
+                return value.replace(/[^0-9a-z-_]/ig, "-");
+            }
+        };
+
+        contentLoaded = true;
+
+        // for IE9 case special handling of del key and backspace key as IE9 doesn"t
```

Comment:
Do we really need a special handling for IE 9 in 2020?

Reply:
I think documenting the oldest supported browsers would help a lot: https://github.com/adobe/aem-core-wcm-components/issues/403